src/Compiler.php

```diff
@@ -252,15 +250,39 @@ protected function parseBindAttributes(string $attributeString): string
 
     protected function attributesToString(array $attributes): string
     {
-        return collect($attributes)
-            ->map(function ($value, string $attribute) {
-                if (is_array($value)) {
-                    $value = '['.$this->attributesToString($value).']';
-                }
+        $attributes = collect($attributes);
+
+        $string = [];
+
+        $string['plain'] = '['.$attributes
+                ->reject(function ($value, string $attribute) {
+                    return Str::startsWith($attribute, '...');
+                })
+                ->map(function ($value, string $attribute) {
+                    if (is_array($value)) {
+                        $value = $this->attributesToString($value);
+                    }
+
+                    return "'{$attribute}' => {$value}";
+                })
+                ->implode(',').']';
+
+        $string['spread'] = $attributes
+                ->filter(function ($value, string $attribute) {
+                    return Str::startsWith($attribute, '...');
+                })
+                ->map(function ($value, string $attribute) {
+                    $attribute = Str::after($attribute, '...');
+
+                    return "\${$attribute}";
```

Comment:
Is there a reason why we dropped the `$` for spread variables? E.g. `<text-field ...input />` instead of  `<text-field ...$input />`?

Reply:
Not a particular one, you are right that it could be easier to see what happens if we keep the `$`. Will change this.

Reply:
Okay, there is a reason: the `$` isn't a valid character in the HTML attributes name and I get an exception:

```
1) Spatie\BladeX\Tests\Features\ComponentCompilation\ComponentCompilationTest::it_compiles_a_attribute_spread_component
PHPUnit\Framework\Exception: 
error parsing attribute name

attributes construct error

Couldn't find end of Start Tag text-field line 1


/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/vendor/spatie/phpunit-snapshot-assertions/src/Drivers/XmlDriver.php:34
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/vendor/spatie/phpunit-snapshot-assertions/src/Snapshot.php:60
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/vendor/spatie/phpunit-snapshot-assertions/src/MatchesSnapshots.php:176
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/vendor/spatie/phpunit-snapshot-assertions/src/MatchesSnapshots.php:56
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/vendor/spatie/phpunit-snapshot-assertions/src/MatchesSnapshots.php:66
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/tests/TestCase.php:48
/Users/tomwitkowski/git/forked/spatie/laravel-blade-x/tests/Features/ComponentCompilation/ComponentCompilationTest.php:210

```

Reply:
Hm, that's fair enough. I really want to keep the `$` in there to keep it really clear we're spreading a variable. We can probably get away with changing the snapshot driver for the tests to plaintext tho.

Reply:
Sorry, was my fault - haven't adjusted the regex patterns why the `$` was part of the compiled view.
Have adjusted them now and now it works.
https://github.com/spatie/laravel-blade-x/pull/112/commits/fc69b51972df1d13a2630d5ebb853e4dc85dd0d1